pandas/tests/arrays/datetimes/test_constructors.py

```diff
@@ -161,3 +164,87 @@ def test_2d(self, order):
         res = DatetimeArray._from_sequence(arr)
         expected = DatetimeArray._from_sequence(arr.ravel()).reshape(arr.shape)
         tm.assert_datetime_array_equal(res, expected)
+
+
+# ----------------------------------------------------------------------------
+# Arrow interaction
+
+
+pyarrow_skip = td.skip_if_no("pyarrow")
+
+
+@pytest.mark.parametrize(
+    ("pa_unit", "pd_unit", "pa_tz", "pd_tz"),
+    [
+        ("s", "s", "UTC", "UTC"),
+        ("ms", "ms", "UTC", "Europe/Berlin"),
+        ("us", "us", "US/Eastern", "UTC"),
+        ("ns", "ns", "US/Central", "Asia/Kolkata"),
+        ("ns", "s", "UTC", "UTC"),
+        ("us", "ms", "UTC", "Europe/Berlin"),
+        ("ms", "us", "US/Eastern", "UTC"),
+        ("s", "ns", "US/Central", "Asia/Kolkata"),
+    ],
+)
+@pyarrow_skip
+def test_from_arrow_with_different_units_and_timezones(pa_unit, pd_unit, pa_tz, pd_tz):
+    import pyarrow as pa
```

Comment:
Since you need the import, you can use `pa = pytest.importorskip("pyarrow")` instead of `pyarrow_skip`